plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -180,6 +181,11 @@ public void testV2TableWithEqualityDelete()
         assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
         // natiokey is before the equality delete column in the table schema, comment is after
         assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
+
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation", 25);
+        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()), Optional.of(new PartitionData(new Long[]{2L})), ImmutableMap.of("regionkey", 2L));
+        // the equality delete file is applied to 2 data files
+        assertQuery("SELECT count(*) FROM \"" + tableName + "$files\" WHERE content = " + FileContent.EQUALITY_DELETES.id(), "VALUES 2");
```

Comment:
Do I understand correctly - we have a duplicated equality delete file in the listing here?

Reply:
We have two equality delete files:

- `regionkey=1` (generated by 1st `writeEqualityDeleteToNationTable` call), which should be applied to the initial data file
- `regionkey=2` (generated by 2nd `writeEqualityDeleteToNationTable` call) which should be applied to two data files(initial create and 2nd insert)

So the equality files count check in `$files` is supposed to return 2, not 3 (`regionkey=2` delete file could be double counted by two data files).

Reply:
If I understand it correctly, I think the `BaseFilesTable` is inherited from `BaseMetadataTable`. It only scans the manifest files (https://github.com/apache/iceberg/blob/3c42a105883ff6edcc3cee71155b59ba89d5146d/core/src/main/java/org/apache/iceberg/BaseFilesTable.java#L91), in which delete files should be unique. The delete file will be planed with data files in the data TableScan to delete all the records with a smaller sequence number(previous commit).